### DISCUSSION TITLE
Does "WithSpan annotation + java agent" has the ability to record the exception and set statusCode error?

### DISCUSSION THREAD

**Discussion by changliu-wk** [open-telemetry/opentelemetry-java-instrumentation#6525] [Q&A] [answered]
If I have a method is using WithSpan annotation
```
otel.library.name
io.opentelemetry.opentelemetry-extension-annotations-1.0
otel.library.version
1.18.0-alpha, 
```
and the latest OTEL Java agent in my application. 
If this method throw exceptions, will the agent automatically set current  span status as ERROR and record this exception?

**Comment by laurit** [comment, marked as answer]
yes

**Reply by changliu-wk** [reply]
> yes

Thanks, Laurit. I thought it has the ability indeed, but it might have something wrong from our side. Where the information is stored? We are using NewRelic as our UI output. We just can see the status is error, the detailed description is empty, unless we explicitly call Span.current().recordException(e), the exception stack trace won't show up in the UI.

**Reply by laurit** [reply]
If you want to verify what exactly is sent the easiest way is to send traces to collector and use `logging/logging_debug` exporter in traces pipeline. You should see something like
```
ScopeSpans #0
ScopeSpans SchemaURL:
InstrumentationScope io.opentelemetry.opentelemetry-extension-annotations-1.0 1.18.0-alpha-SNAPSHOT
Span #0
    Trace ID       : 7f23f611d83b04c88203d3e92b12abf9
    Parent ID      :
    ID             : ff902d8233c46a82
    Name           : Main.test
    Kind           : SPAN_KIND_INTERNAL
    Start time     : 2022-09-01 12:19:07.340398 +0000 UTC
    End time       : 2022-09-01 12:19:07.344175676 +0000 UTC
    Status code    : STATUS_CODE_ERROR
    Status message :
Attributes:
     -> thread.name: STRING(main)
     -> code.function: STRING(test)
     -> thread.id: INT(1)
     -> code.namespace: STRING(Main)
Events:
SpanEvent #0
     -> Name: exception
     -> Timestamp: 2022-09-01 12:19:07.341758342 +0000 UTC
     -> DroppedAttributesCount: 0
     -> Attributes:
         -> exception.message: STRING(foo)
         -> exception.stacktrace: STRING(java.lang.RuntimeException: foo
	at Main.test(Main.java:15)
	at Main.main(Main.java:7)
)
         -> exception.type: STRING(java.lang.RuntimeException)
```
There is also a logging exporter in java agent but this will not print out span events so it can't be use to see the exception.
If you believe that the behaviour you are observing is a bug then please provide a minimal application that reproduces the issue.

**Reply by changliu-wk** [reply]
Thanks! I will verify it later!

**Reply by changliu-wk** [reply]
@laurit, we tried **`otel.traces.exporter = logging/logging_debug`** and **`otel.traces.exporter = logging_debug`**, both are not recognized by agent. We tried `otel.traces.exporter = logging`, but the print out is not like you posted:
Mine likes this:
[otel.javaagent 2022-09-20 17:43:38:619 +0000] [http-nio-9762-exec-3] INFO io.opentelemetry.exporter.logging.LoggingSpanExporter - 'Wo11yServiceRequestHandler.sendRest' : * INTERNAL [tracer: io.opentelemetry.opentelemetry-instrumentation-annotations-1.16:1.18.0-alpha] AttributesMap{data={thread.name=http-nio-9762-exec-3, thread.id=33, code.function=sendRest, code.namespace=*Wo11yServiceRequestHandler}, capacity=128, totalAddedValues=4}
How to set up the debug mode? Thanks!

**Reply by laurit** [reply]
To get that kind of output you'll have to send the traces to the collector and use `logging/logging_debug` exporter in traces pipeline. Logging exporter in agent does not print out span events.

**Reply by changliu-wk** [reply]
@laurit, I followed [this page's instruction](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/examples/demo), and changed otel-collector-config.yaml, 
```
service:
  extensions: [pprof, zpages, health_check]
  pipelines:
    traces:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging/logging_debug]
```
But after run `docker-compose up -d`, in the docker logs it shows:
```
Error: failed to get config: invalid configuration: pipeline "traces" references exporter "logging/logging_debug" which does not exist
2022/09/22 11:59:49 collector server run finished with error: failed to get config: invalid configuration: pipeline "traces" references exporter "logging/logging_debug" which does not exist
```

**Reply by changliu-wk** [reply]
I also tried `logging_debug` with the same error.

**Reply by laurit** [reply]
Sorry I didn't realize that logging/logging_debug wasn't a builtin thing. Try

```
exporters:
  logging/logging_debug:
    loglevel: debug
        
service:
  pipelines:
    traces:
      receivers: [ otlp ]
      processors: [ batch ]
      exporters: [ logging/logging_debug ]
```

**Reply by changliu-wk** [reply]
Thanks @laurit, I am able to see it now. 😄
